daml-foundations/daml-ghc/src/DA/Daml/GHC/Compiler/Convert.hs

```diff
@@ -240,6 +240,23 @@ convertInt64 x
     | otherwise =
         unsupported "Int literal out of bounds" (negate x)
 
+convertRational :: Integer -> Integer -> ConvertM LF.Expr
+convertRational num denom
+    -- the _reduced_ denominator needs to be smaller or equal to 10^10.
```

Comment:
Is this really the only constraint? Doesn't the numerator have to fit in some number of its?